$OpenBSD: patch-src__image_wrapper_cpp,v 1.1 2018/11/02 23:32:44 daniel Exp $

Backport commit 7111817 to fix the build with clang on i386.

This is PR #10698

Index: src/_image_wrapper.cpp
--- src/_image_wrapper.cpp.orig
+++ src/_image_wrapper.cpp
@@ -360,13 +360,12 @@ static PyObject *image_pcolor(PyObject *self, PyObject
     numpy::array_view<const float, 1> x;
     numpy::array_view<const float, 1> y;
     numpy::array_view<const agg::int8u, 3> d;
-    unsigned int rows;
-    unsigned int cols;
+    npy_intp rows, cols;
     float bounds[4];
     int interpolation;
 
     if (!PyArg_ParseTuple(args,
-                          "O&O&O&II(ffff)i:pcolor",
+                          "O&O&O&nn(ffff)i:pcolor",
                           &x.converter,
                           &x,
                           &y.converter,
@@ -404,13 +403,12 @@ static PyObject *image_pcolor2(PyObject *self, PyObjec
     numpy::array_view<const double, 1> x;
     numpy::array_view<const double, 1> y;
     numpy::array_view<const agg::int8u, 3> d;
-    unsigned int rows;
-    unsigned int cols;
+    npy_intp rows, cols;
     float bounds[4];
     numpy::array_view<const agg::int8u, 1> bg;
 
     if (!PyArg_ParseTuple(args,
-                          "O&O&O&II(ffff)O&:pcolor2",
+                          "O&O&O&nn(ffff)O&:pcolor2",
                           &x.converter_contiguous,
                           &x,
                           &y.converter_contiguous,
